internal/data/kv_pair.go

```diff
@@ -377,6 +377,77 @@ func (rp kvPairRepo) CreateCotaEntryKvPairs(ctx context.Context, checkInfo biz.C
 				return err
 			}
 		}
+
+		if kvPair.HasSubKeyPairs() {
+			subKeyPairs := make([]SubKeyKvPair, len(kvPair.SubKeyPairs))
+			subKeyPairVersions := make([]SubKeyKvPairVersion, len(kvPair.ExtensionPairs))
+			for i, subKey := range kvPair.SubKeyPairs {
+				subKeyPairs[i] = SubKeyKvPair{
+					BlockNumber: subKey.BlockNumber,
+					LockHash:    subKey.LockHash,
+					SubType:     subKey.SubType,
+					ExtData:     subKey.ExtData,
+					AlgIndex:    subKey.AlgIndex,
+					PubkeyHash:  subKey.PubkeyHash,
+				}
+				subKeyPairVersions[i] = SubKeyKvPairVersion{
+					BlockNumber: subKey.BlockNumber,
+					LockHash:    subKey.LockHash,
+					SubType:     subKey.SubType,
+					ExtData:     subKey.ExtData,
+					AlgIndex:    subKey.AlgIndex,
+					PubkeyHash:  subKey.PubkeyHash,
+					ActionType:  0,
+				}
+			}
+			if err := tx.Debug().WithContext(ctx).Create(subKeyPairs).Error; err != nil {
+				return err
+			}
+			if err := tx.Debug().WithContext(ctx).Create(subKeyPairVersions).Error; err != nil {
+				return err
+			}
+		}
+		if kvPair.HasUpdatedSubKeyPairs() {
+			updatedSubKeyPairVersions := make([]SubKeyKvPairVersion, len(kvPair.UpdatedSubKeyPairs))
+			updatedSubKeyPairs := make([]SubKeyKvPair, len(kvPair.UpdatedSubKeyPairs))
+			for i, subKey := range kvPair.UpdatedSubKeyPairs {
+				var oldSubKey SubKeyKvPair
+				if err := tx.Model(SubKeyKvPair{}).WithContext(ctx).Where("`lock_hash` = ?", subKey.LockHash).First(&oldSubKey).Error; err != nil {
```

Comment:
Each lock hash could have many subkeys whoes pubkey hashes are not same.

Reply:
Ok, Updated